Tests/BankCardNumberTextInputFormatterTests.swift

```diff
@@ -0,0 +1,175 @@
+//
+//  BankCardNumberTextInputFormatterTests.swift
+//  TextInputKit
+//
+//  Created by Artem Starosvetskiy on 02/12/2016.
+//  Copyright © 2016 Artem Starosvetskiy. All rights reserved.
+//
+
+import XCTest
+@testable import TextInputKit
+
+private func textInputSimulator() -> TextInputSimulator {
+    return TextInputSimulator(BankCardNumberTextInputFormatter(.options()))
+}
+
+class BankCardNumberTextInputFormatterTests: XCTestCase {
+
+    func testThatUatpCardIsFormattedCorrectly() {
+        let textInput = textInputSimulator()
+
+        textInput.insert("122000000000003")
+        textInput.expect("1220 00000 000003", "", "")
+    }
+
+    func testThatAmexCardIsFormattedCorrectly() {
+        let textInput = textInputSimulator()
+
+        textInput.insert("378282246310005")
+        textInput.expect("3782 822463 10005", "", "")
+    }
+
+    func testThatVisaCardIsFormattedCorrectly() {
+        let textInput = textInputSimulator()
+
+        textInput.insert("4111111111111111")
+        textInput.expect("4111 1111 1111 1111", "", "")
+    }
+
+    func testThatMasterCardCardIsFormattedCorrectly() {
+        let textInput = textInputSimulator()
+
+        textInput.insert("5454545454545454")
+        textInput.expect("5454 5454 5454 5454", "", "")
+    }
+
+    func testThatTypingIsFormattedCorrectly() {
+        let textInput = textInputSimulator()
+
+        textInput.insert("3")
```

Comment:
Maybe it's more appropriate to declare tuple or easy structure with {insert: expected:} fields, create array of them and then perform test on each element successively?